tests/test_choices_new_api.py

```diff
@@ -0,0 +1,398 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from unittest import TestCase
+from unittest.mock import Mock
+
+from travertino.colors import NAMED_COLOR, rgb
+from travertino.constants import GOLDENROD, NONE, REBECCAPURPLE, TOP
+from travertino.declaration import BaseStyle, Choices, validated_property
+
+_DATACLASS_ARGS = (
+    {"kw_only": True} if "kw_only" in dataclass.__kwdefaults__ else {"init": False}
+)
+
+
+def prep_style_class(cls):
+    """Decorator to apply dataclass and mock a style class's apply method."""
+    orig_init = cls.__init__
+
+    def __init__(self, *args, **kwargs):
+        self.apply = Mock()
+        orig_init(self, *args, **kwargs)
+
+    cls.__init__ = __init__
+    return dataclass(**_DATACLASS_ARGS)(cls)
+
+
+class PropertyChoiceTests(TestCase):
```

Comment:
We're generally moving towards Pytest as a testing API; if we're adding new tests, I'd rather see those tests written natively using pytest syntax.

Reply:
I've gone through and applied a very literal / naïve translation to pytest so far. It could almost certainly leverage pytest's capabilities better... in fact I think it should probably be possible to supply old- and new-style subclass definitions as parameters to avoid duplicating tests in the first place.

Reply:
I've successfully combined the older and newer test_declarations, using parametrization. Pro: any tests added or modified don't need to be kept in sync in two places for whatever length of time we keep the backwards compatibility. Con: the deprecation tests aren't neatly contained all in one file.

Personally I'm leaning toward it being a net improvement; what do you think?

Reply:
Case in point: I just added a test verifying that a style can't be initialized with a positional argument, and I'm glad I didn't have to do it in two places.